pkg/operator/ceph/cluster/key_rotation.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2025 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cluster to manage a Ceph cluster.
+package cluster
+
+import (
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/ceph/config"
+	"github.com/rook/rook/pkg/operator/ceph/config/keyring"
+	"github.com/rook/rook/pkg/operator/ceph/reporting"
+	"github.com/rook/rook/pkg/operator/ceph/version"
+)
+
+// ClusterKeysToRotate defines the type of ceph daemon keys that need rotation
+type ClusterKeysToRotate struct {
+	RBDMirrorPeer bool
+}
+
+// GetKeysToRotate returns the KeysToRotate struct that defines the type of keys that need rotation.
+func (c *cluster) GetKeysToRotate() (*ClusterKeysToRotate, error) {
+	cephObj := &cephv1.CephCluster{}
+	if err := c.context.Client.Get(c.ClusterInfo.Context, c.namespacedName, cephObj); err != nil {
+		return nil, errors.Wrapf(err, "failed to get cluster %v to get the cephx keys to rotate.", c.namespacedName)
+	}
+
+	// if its a new cluster, that is, the cephXStatus is not initialized yet, then lets initialize it to `Uninitialized` and return
+	if cephObj.Status.CephX == nil {
```

Comment:
Something I realized today reviewing CSI: each key along with CephCluster has a different place in the reconcile, and prior keys will affect subsequent keys. We probably need to make sure each entry in `cephcluster.status.cephx.<entry>` is a pointer that we can check for nil individually.

This would become something like `if cephObj.Status.CephX == nil || cephObj.Status.CephX.RBDMirrorPeer == nil` after the change